File: in_haskell/day_01_part_1.hs
#!/usr/bin/env stack --resolver lts-14.14 runghc

parseInt :: String -> Int
parseInt x = read $ filter (`elem` "-0123456789") x

loadInput :: IO ([Int])
loadInput = do
  raw <- readFile "../inputs/day01.txt"
  return $ map parseInt $ lines raw

massToFuel :: Int -> Int
massToFuel m = (m `div` 3) - 2

main :: IO ()
main = do
  input <- loadInput
  putStr "Day 01 - Part 1: "
  putStrLn $ show $ foldl (+) 0 $ map massToFuel input

File: in_clojure/day_01_part_1.clj
(require '[clojure.string :as str])

(->> "../inputs/day01.txt"
     (slurp)
     (str/split-lines)
     (map read-string)
     (map #(- (quot % 3) 2))
     (reduce +)
     (str "Day 01 - Part 1: ")
     (println))
